### ISSUE TITLE
[image_picker] Clean up PickedFile and related methods.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#144286]
The `image_picker` plugin used to use a precursor of `cross_file` to return multi-platform representation of files: `PickedFile`.

All the methods that return `PickedFile` were removed from the source a while ago, but the type and methods that use them still exist in the package:

* https://github.com/search?q=repo%3Aflutter%2Fpackages+PickedFile&type=code&p=1

We should clean-up `PickedFile` from everywhere.

Affected packages:

* User documentation (README), [publically exports the type](https://github.com/flutter/packages/blob/7cdcf30f8f9919dbf02865fe0c665782bf22360a/packages/image_picker/image_picker/lib/image_picker.dart#L15)
  - [ ] `image_picker`
* Contains code that uses the type:
  - [ ] `image_picker_android`
  - [ ] `image_picker_ios`
  - [ ] `image_picker_macos`
  - [ ] `image_picker_linux`
  - [ ] `image_picker_windows`
* Defines the type, and uses it in interface methods:
  - [ ] `image_picker_platform_interface`

Even though this is deprecating a publically exposed type, this probably *should* not be a breaking change (we bit that bullet on the v1.0 release: https://github.com/flutter/packages/pull/4285)

**Comment by stuartmorgan** [comment]
> [publically exports the type](https://github.com/flutter/packages/blob/7cdcf30f8f9919dbf02865fe0c665782bf22360a/packages/image_picker/image_picker/lib/image_picker.dart#L15)

That's an unfortunate mistake; I missed that when removing the methods.

> Even though this is deprecating a publically exposed type, this probably _should_ not be a breaking change

Unfortunately there's no guarantee that people aren't choosing to use that type in their own code still, even if our own use was removed. I don't see how we can remove it without a breaking change.

It would definitely be a breaking change in the other packages, especially the platform interface since it would break any third-party implementations.

If we do this, we can do https://github.com/flutter/flutter/issues/105800 at the same time.

**Comment by ditman** [comment]
Yeah, this is technically a breaking change... what a bummer. Anyway, that'll make the release "easier" for us too.
